fastapi/param_functions.py

```diff
@@ -2274,12 +2277,14 @@ async def read_items(commons: Annotated[dict, Depends(common_parameters)]):
         return commons
     ```
     """
-    return params.Depends(dependency=dependency, use_cache=use_cache)
+    depends_any: Any = params.Depends(dependency=dependency, use_cache=use_cache)
+    depends: DependencyResult = depends_any
+    return depends
```

Comment:
Why can't we keep the old return login unchanged? I believe the type annotation on the method itself should be enough.

Reply:
return params.Depends(dependency=dependency, use_cache=use_cache)
Fails mypy type checking, because `Depends` is not assignable to the arbitrary `DependencyResult`.

    depends_any: Any = params.Depends(dependency=dependency, use_cache=use_cache)
    return depends_any
Would pass mypy, but not the stronger lints the CI has, failing due to trying to return an `Any` when an `Any` isn't expected.

This whole thing is a bit of a hack, but so is the way `Depend` is used.

Reply:
```py
    depends: Any = params.Depends(dependency=dependency, use_cache=use_cache)
    depends: DependencyResult
    return depends
```

Do you know if that works? It follows that idea but skips the assignment.

Reply:
Easy enough to test.

```
fastapi/param_functions.py:2281: error: Name "depends" already defined on line 2280  [no-redef]
fastapi/param_functions.py:2282: error: Returning Any from function declared to return "DependencyResult"  [no-any-return]
```

Reply:
Thank you. I have one last thought.

> Would pass mypy, but not the stronger lints the CI has, failing due to trying to return an Any when an Any isn't expected.

But do linters check external libraries and complain on them? This project does not have strict type requirements (I think). So is there actually any benefit to fixing the behavior here? Or does it only affect FastAPI contributors?

Reply:
`The CI mentioned is FastAPI's own lint check. Although I was a bit misguided on that comment - it does fail mypy when run with FastAPI's configuration. I just had laxer configuration loaded somehow.

Reply:
You can do as followed:
```
    return cast(
        DependencyResult, params.Depends(dependency=dependency, use_cache=use_cache)
    )
```

Reply:
Good idea, I made the change.